src/ring_theory/subring.lean

```diff
@@ -177,20 +178,20 @@ theorem closure_subset_iff (s t : set R) [is_subring t] : closure s ⊆ t ↔ s
 theorem closure_mono {s t : set R} (H : s ⊆ t) : closure s ⊆ closure t :=
 closure_subset $ set.subset.trans H subset_closure
 
-lemma image_closure {S : Type*} [ring S] (f : R →+* S) (s : set R) :
-  f '' closure s = closure (f '' s) :=
-le_antisymm
-  begin
-    rintros _ ⟨x, hx, rfl⟩,
-    apply in_closure.rec_on hx; intros,
-    { rw [f.map_one], apply is_submonoid.one_mem },
-    { rw [f.map_neg, f.map_one],
-      apply is_add_subgroup.neg_mem, apply is_submonoid.one_mem },
-    { rw [f.map_mul],
-      apply is_submonoid.mul_mem; solve_by_elim [subset_closure, set.mem_image_of_mem] },
-    { rw [f.map_add], apply is_add_submonoid.add_mem, assumption' },
-  end
-  (closure_subset $ set.image_subset _ subset_closure)
+-- lemma image_closure {S : Type*} [ring S] (f : R →+* S) (s : set R) :
+--   f '' closure s = closure (f '' s) :=
+-- le_antisymm
+--   begin
+--     rintros _ ⟨x, hx, rfl⟩,
+--     apply in_closure.rec_on hx; intros,
+--     { rw [f.map_one], apply is_submonoid.one_mem },
+--     { rw [f.map_neg, f.map_one],
+--       apply is_add_subgroup.neg_mem, apply is_submonoid.one_mem },
+--     { rw [f.map_mul],
+--       apply is_submonoid.mul_mem; solve_by_elim [subset_closure, set.mem_image_of_mem] },
+--     { rw [f.map_add], apply is_add_submonoid.add_mem, assumption' },
+--   end
+--   (closure_subset $ set.image_subset _ subset_closure)
```

Comment:
What's wrong with this lemma?

Reply:
My mistake. It was broken at some point in my refactor, but it works again, I just neglected to restore it.